cql-parser/src/main/scala/troy/cql/parser/CqlParser.scala

```diff
@@ -53,7 +55,13 @@ object CqlParser extends JavaTokenParsers
    */
   def constant: Parser[Constant] = {
     import Constants._
-    (string | uuid | number | boolean) ^^ Constant // | hex // TODO
+    def str = (string | uuid) ^^ StringConstant
+    def int = integer ^^ { s => new IntegerConstant(s.toInt) }
+    def floatNum = float ^^ { s => new FloatConstant(s.toFloat) }
+    //def uuid = uuid ^^ { s => new UuidConstant(new UUID (UUID.fromString(s))) }
```

Comment:
There is some magic conversion happening here, `uuid` is a  `Regex`, but you can call any parser methods on it, as if it is a `Parser[String]` (IntellJ should underline the `^^` operator to denote that there is implicit conversation happened). 
However there 2 issues in the code:
1. `def uuid = uuid`, you choose exactly the same name! so you masked the original uuid. 
2. No need for `new UUID` part, `UUID.fromString` return the UUID instance you want.